garage/tf/envs/base.py

```diff
@@ -32,29 +19,26 @@ def __call__(self, *args, **kwargs):
             self.env_cls(*args, **dict(self.extra_kwargs, **kwargs)))
 
 
-class TfEnv(ProxyEnv):
-    @cached_property
-    def observation_space(self):
-        return to_tf_space(self.wrapped_env.observation_space)
+class TfEnv(GarageEnv):
+    """
+    Returns a TensorFlow wrapper class for gym.Env.
 
-    @cached_property
-    def action_space(self):
-        return to_tf_space(self.wrapped_env.action_space)
+    Args:
+        env (gym.Env): the env that will be wrapped
+    """
 
-    @cached_property
-    def spec(self):
-        return EnvSpec(
-            observation_space=self.observation_space,
-            action_space=self.action_space,
-        )
+    def __init__(self, env):
+        super(TfEnv, self).__init__(env)
+        self.action_space = self._to_tf_space(self.action_space)
```

Comment:
`self.action_space` and `self.observation_space` should only ever take on one type. why not `self._to_garage_space(env.action_space)`